tests/test_git.py

```diff
@@ -706,6 +706,95 @@ def test_fetch_from_file(self):
             self.assertEqual(commit['category'], 'commit')
             self.assertEqual(commit['tag'], 'http://example.com.git')
 
+    def test_fetch_recovery(self):
```

Comment:
The test looks great but I think you can improve it splitting in on different cases and not having everything tested on the same one.